src/app/profile/Profile.tsx

```diff
@@ -24,10 +33,12 @@ export const Profile = ({ profileDetails, isLoading, isError, pageTitle }: Profi
     );
   }
 
-  if (isError || !profileDetails || !pageTitle) {
+  if (!!error || !profileDetails || !pageTitle) {
     return (
       <Card>
-        <Typography align="center">{formatMessage({ id: 'error' })}</Typography>
+        <Typography align="center">
+          {String(error).includes('404') ? formatMessage({ id: 'error' }) : formatMessage({ id: 'error' })}
```

Comment:
if error else error... ciekawe ;)
W ogóle, jeśli i tak jest redirect, to można tutaj chamsko zwracać `null`a